src/overmap_ui.cpp

```diff
@@ -82,6 +84,68 @@ std::tuple<char, nc_color, size_t> get_note_display_info( const std::string &not
     return result;
 }
 
+std::array<std::pair<nc_color, long>, 9> get_overmap_neighbors( const tripoint &current )
+{
+    const bool has_debug_vision = g->u.has_trait( trait_id( "DEBUG_NIGHTVISION" ) );
+
+    std::array<std::pair<nc_color, long>, 9> map_around;
+    int index = 0;
+    for( const tripoint &dest : g->m.points_in_radius( current, 1 ) ) {
+        nc_color ter_color = c_black;
+        long ter_sym = ' ';
+        const bool see = has_debug_vision || overmap_buffer.seen( dest.x, dest.y, dest.z );
+        if( see ) {
+            // Only load terrain if we can actually see it
+            oter_id cur_ter = overmap_buffer.ter( dest );
+            ter_color = cur_ter->get_color();
+            ter_sym = cur_ter->get_sym();
+        } else {
+            ter_color = c_dark_gray;
+            ter_sym   = '#';
+        }
+        map_around[index++] = std::make_pair( ter_color, ter_sym );
+    }
+    return map_around;
+}
+
+void update_note_preview( const std::string &note,
+                          const std::array<std::pair<nc_color, long>, 9> map_around,
+                          const std::tuple<catacurses::window *, catacurses::window *, catacurses::window *>
+                          &preview_windows )
+{
+    auto om_symbol = get_note_display_info( note );
+    const nc_color note_color = std::get<1>( om_symbol );
+    const char symbol = std::get<0>( om_symbol );
+    const std::string note_text = note.substr( std::get<2>( om_symbol ), std::string::npos );
+
+    auto w_preview       = std::get<0>( preview_windows );
+    auto w_preview_title = std::get<1>( preview_windows );
+    auto w_preview_map   = std::get<2>( preview_windows );
+
+    draw_border( *w_preview );
+    mvwprintz( *w_preview, 1, 1, c_white, _( "Note preview" ) );
+    wrefresh( *w_preview );
+
+    werase( *w_preview_title );
+    mvwprintz( *w_preview_title, 0, 0, c_yellow, note_text );
+    mvwputch( *w_preview_title, 0, note_text.length(), c_white, LINE_XOXO );
+    for( size_t i = 0; i < note_text.length(); i++ ) {
+        mvwputch( *w_preview_title, 1, i, c_white, LINE_OXOX );
+    }
+    mvwputch( *w_preview_title, 1, note_text.length(), c_white, LINE_XOOX );
+    wrefresh( *w_preview_title );
+
+    draw_border( *w_preview_map, c_yellow );
+    for( int i = 1; i < 4; i++ ) {
```

Comment:
Starting from `0` would be more conventional and handy in case of named constants.